src/cargo/ops/cargo_add/mod.rs

```diff
@@ -124,11 +126,71 @@ pub fn add(workspace: &Workspace<'_>, options: &AddOptions<'_>) -> CargoResult<(
                 inherited_features.iter().map(|s| s.as_str()).collect();
             unknown_features.extend(inherited_features.difference(&available_features).copied());
         }
+
         unknown_features.sort();
+
+        let mut activated: IndexSet<_> =
+            dep.features.iter().flatten().map(|s| s.as_str()).collect();
+        if dep.default_features().unwrap_or(true) {
+            activated.insert("default");
+        }
```

Comment:
Is this dead code?

Reply:
yes 👍🏼 if only clippy could be able to catch that case